prow/github/hmac.go

```diff
@@ -46,3 +64,37 @@ func PayloadSignature(payload []byte, key []byte) string {
 	sum := mac.Sum(nil)
 	return "sha1=" + hex.EncodeToString(sum)
 }
+
+// For given repository/organization, returns all *valid* HMAC tokens. Validity is determined based on time till they are valid.
+func ExtractHmacs(repo string, tokenGenerator func() []byte) [][]byte {
+	t := tokenGenerator()
+	repoToTokenMap := map[string]map[string]time.Time{}
+
+	err := yaml.Unmarshal(t, &repoToTokenMap)
+	if err != nil {
+		fmt.Printf("error: %v", err)
+		return nil
+	}
+
+	orgName := strings.Split(repo, "/")[0]
+	var extractValidToken = func(allTokens map[string]time.Time) [][]byte {
+		var validTokens [][]byte
+		for key, value := range allTokens {
+			if value.After(time.Now()) {
+				validTokens = append(validTokens, []byte(key))
+			}
+		}
+		return validTokens
+	}
+
+	if val, ok := repoToTokenMap[repo]; ok {
+		return extractValidToken(val)
+	}
+	if val, ok := repoToTokenMap[orgName]; ok {
+		return extractValidToken(val)
+	}
+	if val, ok := repoToTokenMap["global_hmac"]; ok {
+		return extractValidToken(val)
+	}
```

Comment:
I wonder whether we should instead permit all the possible for any parent - for instance, if we're migrating from repo-level webhooks to org-level webhooks, we would need to add the new key for every existing repo as part of the migration. We could avoid that with something like this:

```suggestion
  var validTokens []byte
	if val, ok := repoToTokenMap[repo]; ok {
		validTokens = append(validTokens, extractValidToken(val)...)
	}
	if val, ok := repoToTokenMap[orgName]; ok {
		validTokens = append(validTokens, extractValidToken(val)...)
	}
	if val, ok := repoToTokenMap["global_hmac"]; ok {
		validTokens = append(validTokens, extractValidToken(val)...)
	}
	return validTokens
```

(I'm not certain this suggestion is a good idea, but no reason it wouldn't be comes to mind. To the best of my knowledge, anyone who can see org-level hooks also has full control over repos - if I'm wrong, that would be a good reason to not do this. I think I may have argued against something like this before, too...)

Reply:
Personally, I am in favor but I think you and/or @cjwagner had concerns? 
https://docs.google.com/document/d/1m3nFfPt95kbe6BQdWp5BrZMSPkBo5mFb2nuMq1OeE24/edit?disco=AAAAI5tP1FU

Reply:
It is not clear to me how migration from repo lvl to org lvl hooks would be simpler in proposed scenario than in current scenario. We can discuss this offline and update here.

Reply:
We discussed offline and decided that this change is not necessary and we will continue with what was discussed in design doc.